syntax = "proto3";

package armonik.api.grpc.v1.partitions;

import "meta.proto";

option csharp_namespace = "Armonik.Api.Grpc.V1.Partitions";

/**
 * A raw partition object.
 *
 * Used when a list or a single partition is returned.
 */
message PartitionRaw {
  string id = 1; /** The partition ID. */
  repeated string parent_partition_ids = 2; /** The parent partition IDs. */
  int64 pod_reserved = 3; /** Whether the partition is reserved for pods. */
  int64 pod_max = 4; /** The maximum number of pods that can be used by sessions using the partition. */
  map<string, string> pod_configuration = 7; /** The pod configuration. */
  int64 preemption_percentage = 5; /** The percentage of the partition that can be preempted. */
  int64 priority = 6; /** The priority of the partition. */
}

/**
 * Request to list partitions.
 */
message ListPartitionsRequest {
  meta.Meta meta = 1; /** The meta information. */
}

/**
 * Response to list partitions.
 *
 * Use pagination, filtering and sorting from the request.
 * Retunr a list of raw partitions.
 */
message ListPartitionsResponse {
  repeated PartitionRaw partitions = 1; /** The list of raw partitions. */

  int32 page = 2; /** The page number. Start at 0. */
  int32 page_size = 3; /** The page size. */
  int32 total = 4; /** The total number of partitions. */


}

/**
 * Request to get a partition.
 */
message GetPartitionRequest {
  string id = 1; /** The partition ID. */
}

/**
 * Response to get a partition.
 *
 * Return a raw partition.
 */
message GetPartitionResponse {
  PartitionRaw partition = 1; /** The raw partition. */
}
